lib/logstash/inputs/kafka.rb

```diff
@@ -133,31 +133,32 @@ def run(logstash_queue)
     @logger.info('Running kafka', :group_id => @group_id, :topic_id => @topic_id, :zk_connect => @zk_connect)
     begin
       @consumer_group.run(@consumer_threads,@kafka_client_queue)
-      begin
-        while true
-          event = @kafka_client_queue.pop
-          queue_event(event, logstash_queue)
-        end
-      rescue LogStash::ShutdownSignal
-        @logger.info('Kafka got shutdown signal')
-        @consumer_group.shutdown
+
+      while !stop?
+        event = @kafka_client_queue.pop
```

Comment:
is there an example elsewhere of how this should be bypassed?. I suppose I could check the size of the queue and whether it should stop
```
continue if @kafka_client_queue.empty?
```